Services/DataProviders/GithubDataSourceAdaptee.cs

```diff
@@ -0,0 +1,190 @@
+/*
+* Digital Excellence Copyright (C) 2020 Brend Smits
+* 
+* This program is free software: you can redistribute it and/or modify 
+* it under the terms of the GNU Lesser General Public License as published 
+* by the Free Software Foundation version 3 of the License.
+* 
+* This program is distributed in the hope that it will be useful, 
+* but WITHOUT ANY WARRANTY; without even the implied warranty 
+* of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+* See the GNU Lesser General Public License for more details.
+* 
+* You can find a copy of the GNU Lesser General Public License 
+* along with this program, in the LICENSE.md file in the root project directory.
+* If not, see https://www.gnu.org/licenses/lgpl-3.0.txt
+*/
+
+using AutoMapper;
+using Microsoft.Extensions.Configuration;
+using Models;
+using Newtonsoft.Json;
+using Repositories;
+using RestSharp;
+using Services.DataProviders.Resources;
+using Services.Sources;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+namespace Services.DataProviders
+{
+
+    public class GithubDataSourceAdaptee : IAuthorizedDataSourceAdaptee, IPublicDataSourceAdaptee
+    {
+        /// <summary>
+        /// A factory that will generate a rest client to make API requests.
+        /// </summary>
+        private readonly IRestClientFactory restClientFactory;
+
+        /// <summary>
+        /// Mapper object from auto mapper that will automatically maps one object to another.
+        /// </summary>
+        private readonly IMapper mapper;
+
+        private readonly string clientSecret;
+        private readonly string clientId;
+
+        public GithubDataSourceAdaptee(
+            IConfiguration configuration,
+            IRestClientFactory restClientFactory,
+            IMapper mapper)
+        {
+            this.restClientFactory = restClientFactory;
+            this.mapper = mapper;
+
+            clientId = configuration.GetSection($"{Title}ClientId").Value;
+            clientSecret = configuration.GetSection($"{Title}ClientSecret").Value;
+            OauthUrl = "https://github.com/login/oauth/authorize?client_id=" + clientId + $"&scope=repo&state={Title}";
```

Comment:
Maybe also fetch this from appsettings? or whatever if it doesnt change its cool.

Reply:
Good idea! Now you mention this, maybe it's even better whenever I retrieve the base URL from the database?

(For the client id and client secret you are completely right, I had that on my to-do list, thanks for the reminder! 😉 )